homeassistant/components/iron_os/__init__.py

```diff
@@ -63,14 +61,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: IronOSConfigEntry) -> bo
     ble_device = bluetooth.async_ble_device_from_address(
         hass, entry.unique_id, connectable=True
     )
-    if not ble_device:
-        raise ConfigEntryNotReady(
-            translation_domain=DOMAIN,
-            translation_key="setup_device_unavailable_exception",
-            translation_placeholders={CONF_NAME: entry.title},
-        )
-
-    device = Pynecil(ble_device)
+
+    device = Pynecil(ble_device or entry.unique_id)
```

Comment:
```suggestion
    device = Pynecil(entry.unique_id)
```
so it might just come down to this